src/vs/workbench/contrib/notebook/browser/services/notebookServiceImpl.ts

```diff
@@ -202,8 +221,36 @@ export class NotebookProviderInfoStore extends Disposable {
 					cellOptions = (options as INotebookEditorOptions | undefined)?.cellOptions;
 				}
 
-				const notebookOptions: INotebookEditorOptions = { ...options, cellOptions, viewState: undefined };
-				const editor = NotebookEditorInput.getOrCreate(this._instantiationService, notebookUri, preferredResource, notebookProviderInfo.id);
+				let notebookOptions: INotebookEditorOptions;
+
+				if (resource.scheme === Schemas.vscodeNotebookCellOutput) {
+					if (!data?.handle) {
+						throw new Error('Invalid cell handle');
+					}
+					const cellUri = data?.notebook.with({ fragment: (data.handle).toString() });
+
+					cellOptions = { resource: cellUri, options };
+
+					const cellIndex = await this._notebookEditorModelResolverService.resolve(notebookUri)
+						.then(model => model.object.notebook.cells.findIndex(cell => cell.uri.fragment === cellUri.fragment))
+						.then(index => index >= 0 ? index : 0);
+
+					const cellIndexesToRanges: ICellRange[] = [{ start: cellIndex, end: cellIndex + 1 }];
+
+					notebookOptions = {
+						...options,
+						cellOptions,
+						viewState: undefined,
+						cellSelections: cellIndexesToRanges
+					};
+				} else {
+					notebookOptions = {
+						...options,
+						cellOptions,
+						viewState: undefined,
+					};
+				}
+				const editor = NotebookEditorInput.getOrCreate(this._instantiationService, notebookUri, undefined, notebookProviderInfo.id);
```

Comment:
we might still need the preferred resource parameter (now undefined in this line) to make the tab label correct in some cases